src/runtime/policy/policy.ts

```diff
@@ -279,6 +290,65 @@ export class PolicyField {
       label: label === 'raw' ? '' : label,
     };
   }
+
+  getRestrictedFields(parentType: Type) {
+    const field = parentType.getEntitySchema().fields[this.name];
+    return [this.name, this.restrictField(field)];
+  }
+
+  private restrictField(field) {
```

Comment:
consider moving these methods into FieldType classes?

Reply:
I thought about it, but don't these require understanding policies? I could refactor it so that I pass the fields to be retained as an argument to `restrictField`. Is that what you are suggesting?

Reply:
I would actually be against that - FieldType is core Arcs stuff while policy analysis is definitely ancillary to that. Any code that is primary for policy analysis should not be inside types library of core arcs.

Reply:
I thought of this as a generic schema / schema fields functionality, without exposing the policy stuff to the type system. When you have 2 similar schemas, and you make a minimal schema out of them. I don't like the word `restrict` either, if you have better naming suggestions, would be great.

Reply:
Ah, I see. So essentially some like `schema.intersect(otherSchema)` as well as `schema.union(otherSchema)` ?

Reply:
So then we could generate schemas from policies, and do `union()` on them.

Reply:
Done.